src/pytorch_lightning/callbacks/stochastic_weight_avg.py

```diff
@@ -118,27 +118,28 @@ def __init__(
         self._annealing_strategy = annealing_strategy
         self._avg_fn = avg_fn or self.avg_fn
         self._device = device
-        self._model_contains_batch_norm = None
-        self._average_model = None
+        self._max_epochs: int
+        self._model_contains_batch_norm: bool
+        self._average_model = pl.LightningModule()
 
     @property
     def swa_start(self) -> int:
-        return max(self._swa_epoch_start - 1, 0)  # 0-based
+        return max(self._swa_epoch_start - 1, 0)  # type: ignore[return-value]
```

Comment:
Please do not remove comments

Reply:
And by the time it is used, it will already have the type properly resolved (no more float, but int only)

Reply:
My bad. The comment has been added back.

Reply:
> And by the time it is used, it will already have the type properly resolved (no more float, but int only)

@otaj I thought so, too, but mypy seems to miss the value range check in `__init__`.

Reply:
it doesn't "miss" anything, the variable is a `Union[int, float]` in `__init__`, it only gets forced to `int` in `on_fit_start` https://github.com/Lightning-AI/lightning/blob/26361adce107234a858de7bd3a163bef43832ffb/src/pytorch_lightning/callbacks/stochastic_weight_avg.py#L149.

But it is first used by `on_train_epoch_start`, so that should be resolved by then. But, mypy doesn't know (and cannot know) about this order

Reply:
So what is the "right" way to resolve this? A simple fix would be put `# type: ignore` there or `assert isinstance(self._swa_epoch_start, int)`.

Reply:
I'd go with assert

Reply:
And check that the assert doesn't break any tests of course :)